src/gcp/apphosting.ts

```diff
@@ -460,3 +516,53 @@ export async function ensureApiEnabled(options: any): Promise<void> {
   const projectId = needProjectId(options);
   return await ensure(projectId, API_HOST, "frameworks", true);
 }
+
+/**
+ * Generates the next build ID to fit with the naming scheme of the backend API.
+ * @param counter Overrides the counter to use, avoiding an API call.
+ * @return
+ */
+export async function getNextRolloutId(
+  projectId: string,
+  location: string,
+  backendId: string,
+  counter?: number,
+): Promise<string> {
+  const date = new Date();
+  const year = date.getUTCFullYear();
+  // Note: month is 0 based in JS
+  const month = String(date.getUTCMonth()).padStart(2, "0");
+  const day = String(date.getUTCDay()).padStart(2, "0");
+
+  if (counter) {
+    return `build-${year}-${month}-${day}-${String(counter).padStart(3, "0")}`;
+  }
+
+  // Note: must use exports here so that listRollouts can be stubbed in tests.
+  const builds = await (exports as { listRollouts: typeof listRollouts }).listRollouts(
```

Comment:
Huh, but `exports.listRollouts` gives you type errors or something? That's unfortunate :(